### PR TITLE
fix(consentOSP): fix VAT ID/identifier in the OSP Consent Form

### PR DESCRIPTION
**Pull request by saadanzari**
## Description

Add missing field, VAT ID/identifier in the OSP Consent Form is the Company details information.
Since the VAT id info is available in the response, it should appear under the Company details section.

## Why

VAT ID is missing on the company details information.

## Issue

https://github.com/eclipse-tractusx/portal-frontend/issues/1323

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally

## Changelog Entry

- **Consent OSP**
  - fixed VAT ID/identifier in the OSP Consent Form, and update roles translations on language switch [#1323](https://github.com/eclipse-tractusx/portal-frontend/issues/1323)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,13 @@
 
 ## Unreleased
 
+## 2.3.0-RC4
+
+### Bugfixes
+
+- **Home Page**
```
Comment (ss-nikunj) [inline CHANGELOG.md:9]: Please remove the changelog entry from the changelog.md file. Could you mention it in the description of your PR?

Comment (saadanzari) [reply CHANGELOG.md]: reverted changelog commits, and updated into the ticket description

File: src/components/pages/OSPConsent/CompanyDetails.tsx [hunk 2]
```diff
@@ -216,6 +223,12 @@ export const CompanyDetails = ({
     )
   }
 
+  const checkUniqueIdType = (type: string): string => {
+    return (
+      uniqueIdTypeText[type as keyof typeof uniqueIdTypeText] ??
+      uniqueIdTypeText.VAT_ID
+    )
+  }
```
Comment (oyo) [inline src/components/pages/OSPConsent/CompanyDetails.tsx:231]: I suggest more compact:

```suggestion
  const checkUniqueIdType = (type: string): string =>
    uniqueIdTypeText[type as keyof typeof uniqueIdTypeText] ??
    uniqueIdTypeText.VAT_ID
```

File: src/components/pages/OSPConsent/CompanyDetails.tsx [hunk 3]
```diff
@@ -227,6 +240,10 @@ export const CompanyDetails = ({
       [t('osp.city'), companyDetails?.city ?? ''],
       [t('osp.region'), companyDetails?.region ?? ''],
       [t('osp.country'), companyDetails?.countryAlpha2Code ?? ''],
+      ...(companyDetails?.uniqueIds?.map((id) => {
+        const strType = checkUniqueIdType(id.type as string)
+        return [strType, id.value]
+      }) ?? []),
```
Comment (oyo) [inline src/components/pages/OSPConsent/CompanyDetails.tsx:246]: I suggest more compact:

```suggestion
      ...(companyDetails?.uniqueIds?.map((id) =>
        [checkUniqueIdType(id.type as string), id.value]
      ) ?? []),
```

**Review by oyo — CHANGES_REQUESTED** [review]
see suggestions

**Pull request merged**